bin/nim-gdb

```diff
@@ -5,17 +5,25 @@ set -e
 
 which nim > /dev/null || (echo "nim not in PATH"; exit 1)
 which gdb > /dev/null || (echo "gdb not in PATH"; exit 1)
+which readlink > /dev/null || which greadlink > /dev/null || (echo "readlink not in PATH. Please install coreutils from brew if on Mac."; exit 1)
+
+nreadlink () {
+    (which greadlink > /dev/null && greadlink "$@") || (which readlink > /dev/null && readlink "$@") || echo "Readlink could not be found"
+}
```

Comment:
Please make a variable with the ``readlink`` command. Something like ``READLINK``. Also please ensure that you are on ``osx`` before printing any troubleshooting guidelines about fixing problems on ``osx``. This you can do with ``uname -s``

Reply:
Not sure if I'm doing it wrong, but a variable doesn't work, only a function will because in NIM_SYSROOT below, the variable expansion leads to bash thinking the expansion is not a command with arguments.

For example,

```bash
READLINK=$((which greadlink > /dev/null && greadlink "$@") || (which readlink > /dev/null && readlink "$@") || echo "Readlink could not be found")

NIM_SYSROOT=$(dirname $(dirname $(READLINK -e $(which nim))))
```

If I run the above script on my Mac, I get:

```
greadlink: missing operand
Try 'greadlink --help' for more information.
READLINK: illegal option -- e
usage: READLINK [-FlLnqrsx] [-f format] [-t timefmt] [file ...]
usage: dirname path
usage: dirname path
```

I will take the suggestion about detecting OS and put it in the followup PR

Reply:
I was thinking about this:
```sh
if [[ $(uname -s) = Darwin]]; then  
  READLINK=greadlink 
  which greadlink > /dev/null || echo "greadlink not in PATH. Please install coreutils from brew."
else
  READLINK=readlink
fi
```
And then you call readlink with ``$READLINK`` instead of ``readlink``.

Reply:
Ah, yep. Looks like that works. I updated the PR. Modified your code a bit to fit in with the code I had written.